src/converse-chatview.js

```diff
@@ -354,6 +354,9 @@
                             'extra_classes': this.getExtraMessageClasses(attrs)
                         })
                     ));
+                    //Where the argument "attrs" come from? Could "attrs" have information from onMessage?
+                    //Could be possible (and would it be ok) to place a button instead of .text(text)?
+                    //Making the button to show the text set here onclick?
```

Comment:
`                    if(typeof attrs.spoiler !== 'undefined' && attrs.spoiler != 'no' ){
                        var button = document.createElement("button");
                        button.type = "button";
                        button.value = attrs.spoiler;
                        button.textContent = attrs.spoiler;

                        button.onclick = function (){
                            alert("Button clicked");
                        };


                        $msg.find('.chat-msg-content').first()
                        //.text(attrs.spoiler)
                        .addHyperlinks()
                        .append(button)
                        .addEmoticons(_converse.visible_toolbar_buttons.emoticons);
                        
                    }else{
                        
                    $msg.find('.chat-msg-content').first()
                        .text(text)
                        .addHyperlinks()
                        .addEmoticons(_converse.visible_toolbar_buttons.emoticons);
                    }
                    return $msg;
                },
`

This is a bit redundant at this point but I'm interested in the
`                        button.onclick = function (){
                            alert("Button clicked");
                        };`

part.

How can I set this function on the button element? Before thinking about creating a specific function for this matter, I wanted to try to create an anonymous function as you can see in the code.
The problem is, in the HTML, there's no onclick attribute. I tried other methods like .on(click.., .bind(, setAttribute, etc. But none of them work.

I guess it is a pretty basic issue, but I'm not able to find the solution.

Thank you :)